hbase-server/src/main/java/org/apache/hadoop/hbase/coprocessor/CoprocessorHost.java

```diff
@@ -144,21 +145,42 @@ protected void loadSystemCoprocessors(Configuration conf, String confKey) {
 
     int currentSystemPriority = Coprocessor.PRIORITY_SYSTEM;
     for (String className : defaultCPClasses) {
-      String[] classNameAndPriority = className.split("\\|");
+      // After HBASE-23710 and HBASE-26714 when configuring for system coprocessor, we accept
+      // an optional format of className|priority|path
+      String[] classNameToken = className.split("\\|");
       boolean hasPriorityOverride = false;
-      className = classNameAndPriority[0];
+      boolean hasPath = false;
+      className = classNameToken[0];
       int overridePriority = Coprocessor.PRIORITY_SYSTEM;
-      if (classNameAndPriority.length > 1){
-        overridePriority = Integer.parseInt(classNameAndPriority[1]);
+      Path path = null;
+      if (classNameToken.length > 1 && !Strings.isNullOrEmpty(classNameToken[1])) {
+        overridePriority = Integer.parseInt(classNameToken[1]);
         hasPriorityOverride = true;
       }
+      if (classNameToken.length > 2 && !Strings.isNullOrEmpty(classNameToken[2])) {
+        path = new Path(classNameToken[2].trim());
+        hasPath = true;
+      }
       className = className.trim();
       if (findCoprocessor(className) != null) {
         // If already loaded will just continue
         LOG.warn("Attempted duplicate loading of " + className + "; skipped");
         continue;
       }
       ClassLoader cl = this.getClass().getClassLoader();
+      // override the class loader if a path for the system coprocessor is provided.
+      if (hasPath) {
+        String pathPrefix = UUID.randomUUID().toString();
+        try {
+          cl = CoprocessorClassLoader.getClassLoader(path, this.getClass().getClassLoader(),
+            pathPrefix, conf);
+        } catch (IOException ioe) {
+          // if system coprocessors cannot be obtained, we also about the region server
+          LOG.error("Cannot fetch external System coprocessor class {} with path {}", className,
+            path);
+          abortServer(className, ioe);
```

Comment:
Can we move this logic under the below try{} catch{}, so that we don't have multiple condition paths for "abortServer(className, ioe);"

Reply:
this is duplicated comments as mentioned as below, resolving it.